nixos/modules/virtualisation/containers.nix

```diff
@@ -318,6 +319,16 @@ let
       '';
     };
 
+    hostPort = mkOption {
+      type = types.nullOr types.string;
```

Comment:
`--port` of `systemd-nspawn` can take a more general form such as`tcp:443:80` (see the following excerpt from systemd-nspawn man page) so for the time being, it's good to be a string than an integer. As for `types.string`, what is the most modern way now?
```
-p, --port=
           If private networking is enabled, maps an IP port on the host onto an IP port on
           the container. Takes a protocol specifier (either "tcp" or "udp"), separated by a
           colon from a host port number in the range 1 to 65535, separated by a colon from
           a container port number in the range from 1 to 65535. The protocol specifier and
           its separating colon may be omitted, in which case "tcp" is assumed. The
           container port number and its colon may be omitted, in which case the same port
           as the host port is implied. This option is only supported if private networking
           is used, such as with --network-veth, --network-zone= --network-bridge=.
```